Grant consent before checking the gtag events

tomalec opened the pull request:
### Changes proposed in this Pull Request:

<!-- You can erase any parts of this template not applicable to your Pull Request. -->
Fixes tests for gtag after consent mode was added.

Adds a plugin with a PHP snippet to grant all params for consent mode, so the E2E tests will have permissions to track events regardless of the environment's region.

As we're planning further improvements to `gtag` implementation (making it closer to GA4W), right now, we'll use this workaround, then we'll add tests suite for `gtag` and consent mode, once the implementation and feature set will be in the final shape.


### Screenshots:

<!--- Optional --->


### Detailed test instructions:
<!-- Add detailed instructions for how to test that this PR fixes the issue and confirm that it doesn't break any other features :) -->

1. checkout, build
2. run E2Es:
```
npm run wp-env:up
npm run test:e2e
```


### Additional details:

<!--
Optional.
Enter a summary of all changes in this Pull Request, which will be added to the changelog if accepted.
Each line should start with change type prefix`(Fix|Add|…) - `, for example:
> Break - A change breaking previous API or functionality.
> Add - A new feature, function or functionality was added.
> Update - Big changes to something that wasn't broken.
> Fix - Took care of something that wasn't working.
> Tweak - Small change, that isn't actually very important.
> Dev - Developer-facing only change.
> Doc - Updated customer or developer facing documentation

If you remove the "Changelog entry" header, the Pull Request title will be used as the changelog entry.

Add the `changelog: none` label if no changelog entry is needed.
-->
### Changelog entry

> Dev - Fix E2E tests for gtags consent mode

tomalec committed:
Disable consent mode for E2E tests

tomalec committed:
Update E2E `trackGtagEvent` to work with pages that does not have any `data` param

tomalec committed:
Use one e2e plugin tfor all snippets

puntope commented on tests/e2e/test-snippets/test-snippets.php:
```diff
@@ -0,0 +1,25 @@
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Snippets;
+
+// Customize/disable the gtag consent mode, to make testing easier by granting everything by default.
+// It's a hack to avoid specifying region for E2E environment, but it tests the customization of consent mode.
```
💅  minor, but maybe we can use a single block comment instead of using // for each line.

tomalec replied:
The way I use comments is:
- block comments `/* */` /`/** */` - for documenting an entity (a module, class, function, etc). 
   More whitespace, and ` * line indention` to catch the attention, for important bits.
- inline comments `//` 
   - for explaining the logic in inline / in between code flow
   - to denote regions/ sections of something that happens
   
  Short and concise not to take too much attention, but flow in line by line with code
   
Like:

```js
/**
 * Checkout module
 *  a module responsible for finishing the purchase...(API/code docs)
 */

/**
 * Checkout form component... (API/code docs)
 */
export class CheckoutForm {
	/**
	 * Submit all the data and finish to create an order...(API/code docs)
	 */
	submit(){
		// Validate data. (explaining flow + section header)
		this.$('.address').validate();
		this.$('.order_notes').validate();

		// Serialize and sanitize (explaining flow + section header)
		let data = this.$('.address').serialize();
		// Server-side expect billing data to be explicitely set,
		// so we fill it using the shipping data as default. (explain "why", flow & details - could happen multi-line)
		data.billing_address = data.billing_address || data.shipping_address;
	}
}
```


Something like the convention described here https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/code-style.html#comments

Problably, it's my habit from the JSDoc days, where `/**` were picked up as autogenerated API docs.

Also, inline comment takes 50% less space ;) which could be good or bad.
IMO, to describe a single function call does not require that much attention to add two more whitespace lines ;) 

-----

Do you suggest some other convention?

puntope replied:
We don't have any convention I guess. I use `//` for 1 line comments and `/**` for block comments. Thats how was intended in C and PHP 

So in this case I would do:

```
/*
* Customize/disable the gtag consent mode, to make testing easier 
* by granting everything by default.
* 
* It's a hack to avoid specifying region for E2E environment, 
* but it tests the customization of consent mode.
*/
```
https://www.php.net/manual/en/language.basic-syntax.comments.php

mikkamp replied:
Not a strict standard we adhere to, but WordPress also recommends to use `//` only for single line comments. https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#5-inline-comments

Not that it doesn't have mixed opinions though: https://make.wordpress.org/core/2022/11/24/proposal-amend-the-inline-documentation-standards-for-multi-line-comments/
I'm comfortable with reading both formats.

tomalec replied:
Thanks for all the links :)
Me and my dyslexia totally relate to the sentiment from https://make.wordpress.org/core/2022/11/24/proposal-amend-the-inline-documentation-standards-for-multi-line-comments/ especially:

> - Reduce possibility of typos and eventual parsing errors ... `/*` to `/**`
> - The de-facto JavaScript coding standard [Prettier](https://prettier.io/)
> - visually very close to the DocBlock standard which may lead to a moment of confusion when reading the code (uh, is this an incomplete DocBlock

But seeing the spiciness of some comments on the WP post and pretty strong opinions from PHP perspective (comparing JS and Prettier conventions _"The recipe instructions for making cheese on toast"_), I'll just hide my set of opinions and struggles and stick to the WP PHPCS rules for JS ;)

tomalec replied:
FWIW, another argument for my convention is that if you select multiple lines in VSCode and press `ctrl + /` you get a nicer result IMHO, especially if you want to comment whole lines without the need select every character. (Plus for some reason `ctrl + shift + a` does not always fork wor me :shrug: )
![image](https://github.com/woocommerce/google-listings-and-ads/assets/17435/164a3f75-d7e0-4180-b972-d1461d4185ff)

I didn't find a command in VSCode to turn selected lines into
```php
/*
 * selected
 * lines
 */
```
do you know one?

puntope commented on tests/e2e/test-snippets/test-snippets.php:
```diff
@@ -0,0 +1,25 @@
+<?php
+/**
+ * Plugin name: E2E test snippets
```
WDYT about naming it as "GLA Test Snippets" to be consistent with the "Gla Test Data" name ?

tomalec replied:
hm.. GLA is quite redundant, as we're in GLA repo, same goes for E2E ;) as that `e2e/`folder, but I' all for consistency :)

puntope reviewed (approved):
Looking good for me @tomalec 

Left some 💅 comments regarding the code.

tomalec committed:
Update Test Snippets plugin name

Addresses https://github.com/woocommerce/google-listings-and-ads/pull/2294#discussion_r1513466096

tomalec committed:
Change comments convention to match WPCS

Address https://github.com/woocommerce/google-listings-and-ads/pull/2294#discussion_r1513444865
